### PR TITLE
Adds AMS package.py

### PR DESCRIPTION
**Pull request by koparasy**
Co-authored-by:  Loic Pottier <pottier1@llnl.gov>, Tom Stitt <stitt4@llnl.gov>

### CODE REVIEW

File: var/spack/repos/builtin/packages/ams/package.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+from spack.package import *
+
+
+class Ams(CMakePackage, CudaPackage):
+    """AMS Autonomous Multiscale Framework."""
+
+    homepage = "https://github.com/LLNL/AMS"
+    git = "git@github.com:LLNL/AMS.git"
+
+    maintainers = ["koparasy", "lpottier"]
```
Comment (lpottier) [inline var/spack/repos/builtin/packages/ams/package.py]: We should use `maintainers()` here to be compatible with Spack 20.x
```suggestion
    maintainers("koparasy", "lpottier")
```

File: var/spack/repos/builtin/packages/ams/package.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ams/package.py]: While you're making changes, please correct the copyright year.

```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

File: var/spack/repos/builtin/packages/ams/package.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+class Ams(CMakePackage, CudaPackage):
+    """AMS Autonomous Multiscale Framework."""
+
+    homepage = "https://github.com/LLNL/AMS"
+    git = "git@github.com:LLNL/AMS.git"
+
+    maintainers = ["koparasy", "lpottier"]
+
+    version("07.25.23-alpha", tag="07.25.23-alpha", submodules=False)
+    version("develop", branch="develop", submodules=False)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/ams/package.py:17]: Please move the branch version to above the tagged one.  It is considered "newer" than numbered versions (see https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison).

**Review by tldahlgren — COMMENTED** [review]
Confirmed the home page, branch, and version tag.  Just some minor nits about copyright and version directive ordering.

File: var/spack/repos/builtin/packages/ams/package.py (hunk 3 above)
Comment (koparasy) [reply var/spack/repos/builtin/packages/ams/package.py:17]: Fixed

File: var/spack/repos/builtin/packages/ams/package.py (hunk 1 above)
Comment (koparasy) [reply var/spack/repos/builtin/packages/ams/package.py]: Fixed

File: var/spack/repos/builtin/packages/ams/package.py (hunk 2 above)
Comment (koparasy) [reply var/spack/repos/builtin/packages/ams/package.py]: Fixed

**Review by lpottier — APPROVED** [review]

**Review by tldahlgren — APPROVED** [review]
Thanks.

**Pull request merged**